### DISCUSSION TITLE
Use list of strings for the --base-branch option on the run command

### DISCUSSION THREAD

**Discussion by mrg2k8** [lindell/multi-gitter#255] [Ideas]
Hello,

I have a set-up with multiple repositories with branches corresponding to various environments. However, some environments might be missing, on purpose:

repo-1, branches dev, uat, main
repo-2, branches uat,main
repo-3, branches main

Moreover, the default branch might not be set correctly.

It would be a good idea to allow a list of strings on the run --base-branch option to try all defined branches in a predefined order and then to default to the default branch if none of these are found. Currently only one branch is being tried.

I can implement this on my own, but wanted to confirm before I start working on it. Thanks.

**Comment by lindell** [comment]
This seems like a good idea!

But I think that this part: "and then to default to the default branch if none of these are found" should be its separate feature/pr/flag since it's not how it works today.

But multiple base branches should be definable without breaking any backwards compatibility 😄

**Comment by mrg2k8** [comment]
It was easier for us to create a script that checks whether the default branch is configured correctly, so I will not be implementing this. We can keep it as an idea for the future.

**Comment by jamestelfer** [comment]
What if it were to support special name of `<default>` in the list of base branches?

**Reply by lindell** [reply]
My first instinct was to like this. But when thinking about it a bit more, I realised that it only ever make sense to have it at the end.
`--base-branches master,main,<default>` doing something like `--base-branches master,<default>,main` will never pick `main`.

Because of this, I think that a flag is better. I don't know the exact name, but something like "use default branch if defined base branch does not exist", but in a shorter form.

**Reply by jamestelfer** [reply]
This would be neat. I'm currently using a convention where we create a `preupdate-fix` branch if we need to, then use some git scripting shenanigans to rebase the multi-gitter branch on that if it exists.

The only odd thing that I currently do is rebase the `preupdate-fix` branch onto the default branch first. I don't think that's functionality that would be all that simple or useful to put into this tool, and it would be a bit error prone.

**Reply by ns-ggeorgiev** [reply]
any progress on support of `<defualt>` option
